com.unity.perception/Runtime/Randomization/Randomizers/SampleRandomizers/Randomizers/ColorRandomizer.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using UnityEngine.Experimental.Perception.Randomization.Parameters;
+using UnityEngine.Experimental.Perception.Randomization.Randomizers.SampleRandomizers.Tags;
+
+namespace UnityEngine.Experimental.Perception.Randomization.Randomizers.SampleRandomizers
+{
+    /// <summary>
+    /// Randomizes the material color of objects tagged with a ColorRandomizerTag
+    /// </summary>
+    [Serializable]
+    [AddRandomizerMenu("Perception/Color Randomizer")]
+    public class ColorRandomizer : Randomizer
+    {
+        static readonly int k_BaseColor = Shader.PropertyToID("_BaseColor");
+
+        /// <summary>
+        /// Describes the range of random colors to assign to tagged objects
+        /// </summary>
+        public ColorHsvaParameter colorParameter;
+
+        /// <summary>
+        /// Randomizes the colors of tagged objects at the start of each scenario iteration
+        /// </summary>
+        protected override void OnIterationStart()
+        {
+            var taggedObjects = tagManager.Query<ColorRandomizerTag>();
+            foreach (var taggedObject in taggedObjects)
+            {
+                var renderer = taggedObject.GetComponent<MeshRenderer>();
+                renderer.material.SetColor(k_BaseColor, colorParameter.Sample());
```

Comment:
Two things I can do now:
1. I'll add a renderer component requirement to the ColorRandomizerTag
2. I'll let the ColorRandomizer look for the Renderer base class instead of a MeshRenderer specifically. That should allow other renderer components with materials be compatible with this randomizer.